server/rest/workspace/list_test.go

```diff
@@ -0,0 +1,104 @@
+		fake    *mocks.MockFakeResponseWriter
+	)
+
+	BeforeEach(func() {
+		ctrl = gomock.NewController(GinkgoT())
+		w = &restworkspacesv1alpha1.Workspace{}
+		w.Name = "foo"
+		w.Namespace = "bar"
+
+		request = buildListRequest(w)
+		fake = mocks.NewMockFakeResponseWriter(ctrl)
+	})
+
+	AfterEach(func() { ctrl.Finish() })
+
+	DescribeTable("workspace GET handler: read",
+		func(
+			mapperFunc workspace.ListWorkspaceMapperFunc,
+			readHandler workspace.ListWorkspaceQueryHandlerFunc,
+			marshaler marshal.MarshalerProvider,
+			responseFunc func() http.ResponseWriter,
+		) {
+			response := responseFunc()
+			handler := workspace.NewListWorkspaceHandler(mapperFunc, readHandler, marshaler)
+			handler.ServeHTTP(response, request)
+		},
+		Entry("failure in marshal provider", workspace.MapListWorkspaceHttp, nopListHandler, errorMarshalProvider, func() http.ResponseWriter {
+			fake.EXPECT().WriteHeader(http.StatusBadRequest)
+			return fake
+		}),
+		Entry("failure in read handler", workspace.MapListWorkspaceHttp, badListHandler, marshal.DefaultMarshalerProvider, func() http.ResponseWriter {
+			fake.EXPECT().WriteHeader(http.StatusInternalServerError)
+			return fake
+		}),
+		Entry("failure marshaling response", workspace.MapListWorkspaceHttp, nopListHandler, badMarshalProvider, func() http.ResponseWriter {
+			fake.EXPECT().WriteHeader(http.StatusInternalServerError)
+			return fake
+		}),
+		Entry("failure to write response", workspace.MapListWorkspaceHttp, nopListHandler, marshal.DefaultMarshalerProvider, func() http.ResponseWriter {
+			fake.EXPECT().Header().Return(http.Header{})
+			fake.EXPECT().Write(gomock.Any()).Return(0, fmt.Errorf("failed to write response body"))
+			fake.EXPECT().WriteHeader(http.StatusInternalServerError)
+			return fake
+		}),
+		Entry("failure to write response", workspace.MapListWorkspaceHttp, nopListHandler, marshal.DefaultMarshalerProvider, func() http.ResponseWriter {
+			fake.EXPECT().Header().Return(http.Header{})
+			fake.EXPECT().Write(gomock.Any()).DoAndReturn(func(a any) (int, error) {
+				slice, ok := a.([]byte)
+				Expect(ok).To(BeTrue())
+				return len(slice), nil
+			})
+			return fake
+		}),
+	)
+})
+
+func badListHandler(ctx context.Context, cmd coreworkspace.ListWorkspaceQuery) (*coreworkspace.ListWorkspaceResponse, error) {
+	return nil, fmt.Errorf("bad create handler")
+}
+
+func nopListHandler(_ctx context.Context, cmd coreworkspace.ListWorkspaceQuery) (*coreworkspace.ListWorkspaceResponse, error) {
+	return &coreworkspace.ListWorkspaceResponse{
+		Workspaces: restworkspacesv1alpha1.WorkspaceList{},
+	}, nil
+}
+
+func buildListRequest(workspace *restworkspacesv1alpha1.Workspace) *http.Request {
+	byteSlice, err := marshal.DefaultMarshal.Marshal(workspace)
+	Expect(err).NotTo(HaveOccurred())
+
+	url := fmt.Sprintf("/apis/workspaces.io/v1alpha1/namespaces/%s/workspaces", workspace.GetNamespace())
+
+	request, err := http.NewRequest("POST", url, bytes.NewReader(byteSlice))
```

Comment:
List requests are `GET`, not `POST`, right?  If so, replace this with `http.MethodGet`.

Reply:
Great catch! May you add the suggestion?

Reply:
```suggestion
	request, err := http.NewRequest(http.MethodGet, url, bytes.NewReader(byteSlice))
```